File: SortingAndSearching/LinearSearch.js
// module imports
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// declarations and definitions
let myInputArray = [];
let result = [];
let foundAtIndex;

// pushing args into array
process.argv.forEach( (val, index) => {
    if(index === 0 || index === 1) return;
    myInputArray.push(Number(val));
});

//get search term
console.log("Enter the search term: ");
rl.on('line', (answer) => {
    result = myInputArray.filter( (value,index) => {
        if (value === Number(answer)) {
            foundAtIndex = index;  // A small side-effect | remove if possible
        }
        return value === Number(answer);
    })
    rl.close();
}).on('close', () => {
    if(result.length === 0){
        console.log("Not Found in the suppplied array!");
    }
    else{
        console.log("Found the element: ",result[0]," at index:",foundAtIndex+1);
    }
});

File: SortingAndSearching/LinearSearch.cpp
#include <iostream>

using namespace std;

// function to perform the linear search
int linearSearch(int search, int myInputArray[], int n){
    for (int i=0;i<n;i++){
        if (myInputArray[i] == search)
        {
            return i; //return the index
        }
    }
    return -1; //return -1 if not found
}

int main(){
    int n;
    cout<<"Enter number of elements"<<endl;
    cin>>n;
    int myInputArray[n];
    cout<<"Enter "<<n<<" integers:"<<endl;
    for(int i=0; i<n;i++){
        cin>>myInputArray[i];
    }
    int search;
    cout<<"Enter element to search:"<<endl;
    cin>>search;
    int result = linearSearch(search, myInputArray, n); //store index of queried element in an integer variable
    if(result != -1){
        cout << "Found Element: " << search << " at index: "<<result+1<<endl;
    }
    else{
        cout<<"Element not found!"<<endl;
    }
    return 0;
}